src/main/logutils.ts

```diff
@@ -110,48 +165,153 @@ const tailFile = (path: string) => {
+    let in_quote = false;
+    let open_lists = 0;
+    let value: any = '';
+
+    // Combat log line always has '<timestamp>  <line>' format,
+    // that is, two spaces between ts and line.
+    const tsIndexEnd = line.indexOf('  ');
+    const timestamp = line.substring(0, tsIndexEnd);
+
+    for (let ptr = tsIndexEnd + 2; ptr < line_len; ptr++) {
+        const c = line.charAt(ptr);
+        if (c === '\n') {
+            break;
+        }
+
+        if (in_quote) {
+            if (c === '"') {
+                in_quote = false;
+                continue;
+            }
+
+        } else {
+            switch (c) {
+            case ',':
+                if (open_lists > 0) {
+                    list_items.at(-1)?.push(value);
+                } else {
+                    args_list.push(value);
+                    if (maxSplits && args_list.length >= maxSplits) {
+                        break;
+                    }
+                }
+
+                value = '';
+                continue;
+
+            case '"':
+                in_quote = true;
+                continue;
+
+            case '[':
+            case '(':
+                list_items.push([]);
+                open_lists++;
+                continue;
+
+            case ']':
+            case ')':
+                if (!list_items.length) {
+                    throw `Unexpected ${c}. No list is open.`;
+                }
+
+                if (value) {
+                    list_items.at(-1)?.push(value);
+                }
+
+                value = list_items.pop();
+                open_lists--;
+                continue;
+            }
+        }
+
+        value += c;
+    }
+
+    if (value) {
+        args_list.push(value)
+    }
+
+    if (open_lists > 0) {
+        throw `Unexpected EOL. There are ${open_lists} open tuples/lists.`
+    }
+
+    return new LogLine(timestamp, args_list)
+}
+
+    /**
  * Handle a line from the WoW log. 
  */
 const handleLogLine = (line: string) => {
-    if (line.includes("ARENA_MATCH_START")) {
-        handleArenaStartLine(line);
-    } else if (line.includes("ARENA_MATCH_END")) {
-        handleArenaStopLine(line);
-    } else if (line.includes("ENCOUNTER_START")) {
-        handleRaidStartLine(line);
-    } else if (line.includes("ENCOUNTER_END")) {
-        handleRaidStopLine(line);
-    } else if (line.includes("ZONE_CHANGE")) {
-        handleZoneChange(line);
-    } else if (line.includes("COMBATANT_INFO")) {
-        handleCombatantInfoLine(line);
-    } else if (line.includes("SPELL_AURA_APPLIED")){
-        handleSpellAuraAppliedLine(line);
+    // Parse line, only until the line token is encountered
+    let logLine = splitLogLine(line, 1)
+
+    // Check if we are interested in this event, and if not, discard it
+    if (interestingCombatLogEvents.indexOf(logLine.args[0]) === -1) {
+        return;
+    }
+
+    // Parse the full line
+    logLine = splitLogLine(line)
+
+    switch (logLine.args[0]) {
```

Comment:
Seems a shame to have to hardcode all the interesting events again in this switch block. Is there some neater syntax we could use here? I'm not aware of one - but maybe you have some ideas.

Reply:
There is, but I wasn't too sure if you'd like it due to it perhaps looking too complex. I'll try it out and you can decide for yourself.

Reply:
Check out lastest commit - it does exactly this. If you don't like it, I'll remove it again :)

Reply:
I like it!